images/hub/z2jh.py

```diff
@@ -3,32 +3,71 @@
 
 Methods here can be imported by extraConfig in values.yaml
 """
+from collections import Mapping
+from functools import lru_cache
 import os
-import sys
+
 import yaml
 
+
+# memoize so we only load config once
+@lru_cache()
+def _load_config():
+    """Load configuration from disk
+
+    Memoized to only load once
+    """
+    cfg = {}
+    for source in ('config', 'secret'):
+        path = f"/etc/jupyterhub/{source}/values.yaml"
+        if os.path.exists(path):
+            print(f"Loading {path}")
+            with open(path) as f:
+                values = yaml.safe_load(f)
+            cfg = _merge_dictionaries(cfg, values)
+        else:
+            print(f"No config at {path}")
+    return cfg
+
+
+def _merge_dictionaries(a, b):
+    """Merge two dictionaries recursively.
+
+    Simplified From https://stackoverflow.com/a/7205107
+    """
+    merged = a.copy()
+    for key in b:
+        if key in a:
+            if isinstance(a[key], Mapping) and isinstance(b[key], Mapping):
+                merged[key] = _merge_dictionaries(a[key], b[key])
+            else:
+                merged[key] = b[key]
+        else:
+            merged[key] = b[key]
+    return merged
+
+
 def get_config(key, default=None):
     """
     Find a config item of a given name & return it
 
     Parses everything as YAML, so lists and dicts are available too
+
+    get_config("a.b.c") returns config['a']['b']['c']
     """
-    path = os.path.join('/etc/jupyterhub/config', key)
-    try:
-        with open(path) as f:
-            data = yaml.safe_load(f)
-            return data
-    except FileNotFoundError:
-        return default
-
-def get_secret(key, default=None):
-    """Get a secret from /etc/jupyterhub/secret"""
-    path = os.path.join('/etc/jupyterhub/secret', key)
-    try:
-        with open(path) as f:
-            return f.read().strip()
-    except FileNotFoundError:
-        return default
```

Comment:
You removed `get_secret`?

Reply:
Yes, get_secret no longer makes sense

Reply:
Ah now it makes sense that it no longer makes sense to me as well =)